odoo/addons/base/models/ir_model.py

```diff
@@ -1259,6 +1250,43 @@ def unlink(self):
 
         return result
 
+    def _process_ondelete(self):
+        """ Process the 'ondelete' of the given selection values. """
+        self = self.filtered(lambda s: s.field_id.store and not s.env[s.field_id.model]._abstract)
+        for selection in self:
+            Model = self.env[selection.field_id.model]
+            field = Model._fields[selection.field_id.name]
+            ondelete = (field.ondelete or {}).get(selection.value)
+            if callable(ondelete):
+                ondelete(selection._get_records())
+            elif not ondelete or ondelete == 'set null':
+                selection._replace_selection(None)
```

Comment:
```suggestion
                selection._get_records().write({field.name: False})
```
because `_replace_selection` is now trivial.